### PR TITLE
[AutoMM] Support Custom Metrics in AutoMM

### PR DESCRIPTION
**Pull request by taoyang1122**
*Issue #, if available:*

*Description of changes:*

- Added the support for custom metrics in MultiModalPredictor.
- The custom metric can be passed to the MultiModalPredictor at the initialization stage or at the evaluation stage.
- The API is consistent with [adding custom metrics in TabularPredictor](https://auto.gluon.ai/stable/tutorials/tabular/advanced/tabular-custom-metric.html).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/utils/metric.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import numpy as np
 from sklearn.metrics import f1_score
 
-from autogluon.core.metrics import get_metric
+from autogluon.core.metrics import Scorer, get_metric
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/utils/metric.py:11]: `Score` looks like unused.

File: multimodal/tests/unittests/others/test_metrics.py [hunk 2]
```diff
@@ -105,3 +112,48 @@ def test_symmetric_hit_rate():
...
+    )
+    scores_by_scorer_eval = predictor_by_name.evaluate(
+        data=dataset.test_df,
+        metrics=[metric_scorer],
+    )
+    scores_by_scorer_init = predictor_by_scorer.evaluate(
+        data=dataset.test_df,
+        metrics=None,
+    )
+    assert scores_by_name[metric_name] == scores_by_scorer_eval[metric_name] == scores_by_scorer_init[metric_name]
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/others/test_metrics.py:159]: Is it safe to asser `scores_by_name[metric_name]==scores_by_scorer_init[metric_name]`? This requires the models in `predictor_by_name` and `predictor_by_score` have exactly the same state. Although use the same time limit for them, can we guarantee the model is the same?

**Review by zhiqiangdon — APPROVED** [review]
LGTM! Congrats for your first PR!
